mlrun/datastore/redis.py

```diff
@@ -33,17 +34,21 @@ def __init__(self, parent, schema, name, endpoint="", secrets: dict = None):
         self.headers = None
 
         self.endpoint = self.endpoint or mlrun.mlconf.redis.url
+        if schema == "ds":
+            datastore_profile = datastore_profile_read(name)
```

Comment:
Currently only a `redis` ds may be returned, but in the future once we add more types of profiles, the returned profile may be of a different type. Before you do further processing, you need to verify that the profile you got back is of the correct type.